azurerm/helpers/azure/app_service.go

```diff
@@ -1605,6 +1619,11 @@ func FlattenAppServiceSiteConfig(input *web.SiteConfig) []interface{} {
 
 	result["scm_type"] = string(input.ScmType)
 	result["ftps_state"] = string(input.FtpsState)
+
+	if input.HealthCheckPath != nil {
+		result["health_check_path"] = *input.HealthCheckPath
+	}
```

Comment:
in general we need to set a value for every key inside a dictionary, which helps us catch when Azure doesn't return a value from the API (otherwise Terraform Core will show a diff between "null" and "empty") e.g.

```suggestion
	healthCheckPath := ""
	if input.HealthCheckPath !=  nil {
		healthCheckPath = *input.HealthCheckPath
	}
	result["health_check_path"] = healthCheckPath
```

however since the rest of this class doesn't, we can ignore this for now

Reply:
The API explicitly supports `null` as the value for this field, so I assumed that was what not setting the field would represent.

Reply:
FWIW I can change the above to explicitly set `result["health_check_path"] = nil` when `input.HealthCheckPath == nil` if that is supported.